### PR TITLE
Issue #41 link rel for coswid

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**

### CODE REVIEW

File: draft-birkholz-rats-corim.md [hunk 1]
```diff
@@ -109,15 +109,15 @@ See Body {{mybody}}.
...
+ | Index | Relationship Type Name | Description                                |
+ +=======+========================+============================================+
+ | 12    | needs                 | links to a module this software needs      |
+ +-------+-----------------------+--------------------------------------------+
+ | 13    | runs-on               | links to a module this software runs       |
+ +-------+-----------------------+--------------------------------------------+
+ | 14    | part-of               | links to a module this software is part of |
+ +-------+-----------------------+--------------------------------------------+
+ | 16    | m-supplemental        | provides additional information if a module|
+  +------+------------------------+-------------------------------------------+
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:122]: This should be a [markdown table](https://www.markdownguide.org/extended-syntax/#tables), so something like:

```
| Index | Relationship Type | Definition |
|---
| 13 | runs-on | links to a CoMID tag describing the module on which this software runs |
...
{: #tbl-new-coswid-to-comid-link-ownership-values title="Link Ownership Values"}
```

File: draft-birkholz-rats-corim.md [hunk 2]
```diff
@@ -99,6 +99,52 @@ See {{-coswid}}.
 
 Privacy Considerations
 
+# Concise SWID Data Definition
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:102]:
```suggestion
# Concise SWID Data Extensions
```

File: draft-birkholz-rats-corim.md [hunk 3]
```diff
@@ -99,6 +99,52 @@ See {{-coswid}}.
...
+
+# 2.7 The link-entry Map
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:104]: Section numbering is created automatically.  At the MD level we only specify the heading nesting level by increasing the number of `#` signs prepended:

```suggestion
## The link-entry Map
```

File: draft-birkholz-rats-corim.md [hunk 4]
```diff
@@ -99,6 +99,52 @@ See {{-coswid}}.
...
+
+The link-entry map has been extended to define new relationship between a CoSWID and
+CoMID(identifies a hardware/firmware module).
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:107]:
```suggestion
The link-entry map is extended to specify relationships between CoSWID and CoMID.
```

File: draft-birkholz-rats-corim.md [hunk 5]
```diff
@@ -99,6 +99,52 @@ See {{-coswid}}.
...
+
+When defining CoSWID to CoMID link entry map the following optional elements
+should not be used:
+ artifact
+ ownership
+ use
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:113]:
```suggestion
When using CoSWID to CoMID relations, the following optional elements of the link-entry map MUST NOT used:

* artifact
* ownership
* use
```

File: concise-swid-tag-ext.cddl [hunk 6]
```diff
@@ -0,0 +1,38 @@
+$$link-extension //= (
+  ? privilege => $privilege,
+)
+
+; expansion of rel values to accomodate comid relationship
+$rel /= m-requires
+$rel /= m-runs-on
+$rel /= m-part-of
+$rel /= m-supplemental
+$rel /= -256..64436 / text
```
Comment (thomas-fossati) [inline concise-swid-tag-ext.cddl:10]: this is already in CoSWID, no need to repeat it.
```suggestion
```

File: draft-birkholz-rats-corim.md [hunk 7]
```diff
@@ -99,6 +99,52 @@ See {{-coswid}}.
...
+ --------|----------------------- |--------------------------------------------|
+ | 12    | m-requires             | The link references a prerequisite module  |
+ |       |                        | tag(s) for installing this software.       |
+ | 13    | m-runs-on              | The link references a module tag that      |
+ |       |                        | this software run on                       |
+ | 14    | m-part-of              | The link references a module tag that this |
+ |       |                        | software is part of                        |
+ | 15    | m-supplemental         | The link references a module tag that the  |
+ |       |                        | referencing tag supplements.               |
+
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:131]: This table should go in the IANA section below.  This section should, for each new relation defined, provide the semantic details.

File: draft-birkholz-rats-corim.md [hunk 8]
```diff
@@ -99,6 +99,52 @@ See {{-coswid}}.
...
+When defining CoSWID to CoMID link entry map the following optional elements
+should not be used:
+ artifact
+ ownership
+ use
+
+# CoSWID Indexed Label Values
+
+# 4.4. Link Rel Values
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:117]: Is this further section really needed?  I think this should be in the context of the section above (i.e., "Link-entry Map")

File: draft-birkholz-rats-corim.md [hunk 9]
```diff
@@ -99,6 +99,52 @@ See {{-coswid}}.
...
+ | 12    | m-requires             | The link references a prerequisite module  |
+ |       |                        | tag(s) for installing this software.       |
+ | 13    | m-runs-on              | The link references a module tag that      |
+ |       |                        | this software run on                       |
+ | 14    | m-part-of              | The link references a module tag that this |
+ |       |                        | software is part of                        |
+ | 15    | m-supplemental         | The link references a module tag that the  |
+ |       |                        | referencing tag supplements.               |
+
+# 4.6 Link Privilege Values
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:132]: Ditto

File: draft-birkholz-rats-corim.md [hunk 10]
```diff
@@ -107,4 +153,13 @@ Security Considerations
 
 See Body {{mybody}}.
 
---- back
+New indexes for Software Tag Link Relationship Values needs to be registered as below:
```
Comment (thomas-fossati) [inline draft-birkholz-rats-corim.md:156]: You also need to add the `trusted`/`untrusted` registration.

File: draft-birkholz-rats-corim.md (hunk 7 above)
Comment (yogeshbdeshpande) [reply draft-birkholz-rats-corim.md:210]: This is how it is documented in CoSWID specification in an exact manner! Hence I replicated the tabular syntax from
Section 4.4 of CoSWID specification!

File: draft-birkholz-rats-corim.md (hunk 8 above)
Comment (yogeshbdeshpande) [reply draft-birkholz-rats-corim.md:196]: This is how it is documented in CoSWID specification in an exact manner! Hence I replicated the tabular syntax from
Section 4.4 of CoSWID specification!

File: draft-birkholz-rats-corim.md (hunk 9 above)
Comment (yogeshbdeshpande) [reply draft-birkholz-rats-corim.md:211]: Not applicable now!

File: draft-birkholz-rats-corim.md (hunk 10 above)
Comment (yogeshbdeshpande) [reply draft-birkholz-rats-corim.md:235]: Not Applicable now!

**Comment by thomas-fossati** [comment]
This is OBE (#47)

**Pull request closed without merging**